temporalio/workflow.py

```diff
@@ -192,11 +206,33 @@ def signal(
     ...
 
 
+@overload
+def signal(
+    *, unfinished_handlers_policy: UnfinishedHandlersPolicy
+) -> Callable[[CallableSyncOrAsyncReturnNoneType], CallableSyncOrAsyncReturnNoneType]:
+    ...
+
+
+@overload
+def signal(
+    *, name: str, unfinished_handlers_policy: UnfinishedHandlersPolicy
```

Comment:
I think you can just add this parameter to the end of the name and dynamic overloads and give it a default value